lib/rubocop/cop/style/double_pipe_equals.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for potential usage of the `||=` operator.
+      #
+      # @example
+      #   # bad
+      #   name = name ? name : 'Bozhidar'
+      #
+      #   # bad
+      #   name = if name
+      #            name
+      #          else
+      #            'Bozhidar'
+      #          end
+      #
+      #   # bad
+      #   unless name
+      #     name = 'Bozhidar'
+      #   end
+      #
+      #   # bad
+      #   name = 'Bozhidar' unless name
+      #
+      #   # good - set name to 'Bozhidar', only if it's nil or false
+      #   name ||= 'Bozhidar'
+      class DoublePipeEquals < Cop
+        MSG = 'Use the double pipe equals operator `||=` instead.'.freeze
+
+        def_node_matcher :ternary_assignment?, <<-PATTERN
+          ({lvasgn ivasgn cvasgn gvasgn} _var
+            (if
+              ({lvar ivar cvar gvar} _var)
+              ({lvar ivar cvar gvar} _var)
+              _))
+        PATTERN
+
+        def_node_matcher :unless_assignment?, <<-PATTERN
+          (if
+            ({lvar ivar cvar gvar} _var) nil
+            ({lvasgn ivasgn cvasgn gvasgn} _var
+              _))
+        PATTERN
+
+        def on_if(node)
+          return unless unless_assignment?(node)
+          add_offense(node)
+        end
+
+        def on_lvasgn(node)
+          return unless ternary_assignment?(node)
+          add_offense(node)
+        end
+
+        alias on_ivasgn on_lvasgn
+        alias on_cvasgn on_lvasgn
+        alias on_gvasgn on_lvasgn
+
+        private
+
+        def autocorrect(node)
+          if ternary_assignment?(node)
+            variable, default = get_variable_and_default_from_ternary(node)
+          else
+            variable, default = get_variable_and_default_from_unless(node)
+          end
+
+          lambda do |corrector|
+            corrector.replace(node.source_range,
+                              "#{variable} ||= #{default.source}")
+          end
+        end
+
+        def get_variable_and_default_from_ternary(node)
```

Comment:
Check out the node extension `IfNode`. You might find named methods that give you the correct node parts there. 🙂 Also naming a method with prefix `get_` will likely generate a RuboCop offense.